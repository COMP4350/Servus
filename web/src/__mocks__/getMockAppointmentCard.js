function getMockAppointmentCard() {
    return {
        buyer: 'Risto',
        provider: 'Arveend',
        service_id: '1001',
        created_at: new Date(100),
        booked_time: new Date(500),
    }
}

export const mockAppointmentCard = getMockAppointmentCard();