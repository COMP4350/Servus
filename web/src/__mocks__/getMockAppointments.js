function getMockAppointments() {
    return [
        {
            buyer: 'Risto',
            provider: 'Arveend',
            service_id: '1001',
            created_at: new Date(100),
            booked_time: new Date(500),
        },
        {
            buyer: 'Arveend',
            provider: 'Risto',
            service_id: '1002',
            created_at: new Date(700),
            booked_time: new Date(1200),
        },
        {
            buyer: 'Caden',
            provider: 'Andy',
            service_id: '1003',
            created_at: new Date(1400),
            booked_time: new Date(1900),
        },
    ]
}

function getMockAppointmentsCookie() {
    return {
        username: 'testusername',
    }
}

export const mockAppointments = getMockAppointments();
export const mockAppointmentsCookie = getMockAppointmentsCookie();