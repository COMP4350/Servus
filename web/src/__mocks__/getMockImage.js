function getMockImage() {
    return {
        ownerUsername: 'riki',
        uploadDate: new Date(2012, 1, 1),
        imageName: 'example_image.jpeg',
        imageData: 'the/link/to/data',
        profilePicture: false,
    };
}

export const mockImage = getMockImage();
