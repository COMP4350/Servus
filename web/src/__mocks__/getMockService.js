function getMockService() {
    return {
        _id: '1001',
        name: "Risto's Painting",
        provider: 'Risto',
        description: 'Risto paints you',
        cost: 100,
        duration: '0100',
        availability: [{ weekday: 1, start_time: '0800', end_time: '1800' }],
        ratings: [{ username: 'arvind', rating: 5 }],
    };
}

export const mockService = getMockService();
