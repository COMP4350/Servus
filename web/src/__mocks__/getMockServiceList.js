function getMockServiceList() {
    return [
        {
            name: "Risto's Painting",
            provider: 'Risto',
            description: 'Risto paints you',
            cost: 100,
            duration: 60,
        },
        {
            name: "Reesto's Painting",
            provider: 'Reesto',
            description: 'Reesto paints yee',
            cost: 120,
            duration: 80,
        },
        {
            name: "Rasta's Painting",
            provider: 'Rasta',
            description: 'Rasta paints yaa',
            cost: 150,
            duration: 90,
        },
    ];
}

export const mockServiceList = getMockServiceList();
