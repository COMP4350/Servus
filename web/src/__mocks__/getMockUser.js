function getMockUser() {
    return {
        username: 'testuser',
        firstName: 'test',
        lastName: 'user',
        password: '123',
        bio: 'im a test user',
    };
}

export const mockUser = getMockUser();
