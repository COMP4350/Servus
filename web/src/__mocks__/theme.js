import { createMuiTheme } from '@material-ui/core/';

export default createMuiTheme({
    palette: {
        primary: {
            main: '#647AA3',
        },
        secondary: {
            main: '#EC5732',
        },
        default: {
            main: '#FFF2EB',
        },
    },
    background: {
        dark: '#151515',
        main: '#272727',
    },
    typographs: {
        fontFamily: ['Roboto'],
    },
});
