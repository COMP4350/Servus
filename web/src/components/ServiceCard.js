import React, { useState } from 'react';
import { Card, makeStyles, Typography, Button } from '@material-ui/core';
import { Chip } from '@material-ui/core';

const useStyles = makeStyles(() => ({
    cardView: {
        width: '100%',
        height: '100%',
        backgroundColor: 'transparent',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
    },
    title: {
        fontSize: 18,
        color: 'white',
        width: 'auto',
    },
    chip: {
        margin: 2,
    },
}));

const formatDuration = duration => {
    const hours = parseInt(duration.charAt(0) + duration.charAt(1));
    const minutes = parseInt(duration.charAt(3) + duration.charAt(4));
    return hours > 0
        ? hours.toString() + (hours > 1 ? ' hours ' : ' hour ')
        : '' + minutes > 0
        ? minutes.toString() + ' minutes ' + minutes
        : '';
};

const ServiceCard = props => {
    const classes = useStyles();
    const serviceTags = props.service.tags ? props.service.tags : [];
    const [expanded, setExpanded] = useState(false);
    return (
        <Card
            onClick={() => setExpanded(!expanded)}
            variant="outlined"
            className={classes.cardView}
            key={props.index}>
            <Typography variant="body1" className={classes.title} align="left">
                {props.service.name}
            </Typography>
            <div>
                {serviceTags?.map((tagData, i) => {
                    return (
                        <Chip
                            key={i}
                            size="small"
                            label={tagData}
                            className={classes.chip}
                            color={props.selected ? 'secondary' : 'default'}
                            disabled={!props.selected}
                        />
                    );
                })}
            </div>
            {expanded ? (
                <div className={classes.expandedInfo}>
                    <Typography>{props.service.description}</Typography>
                    <Typography>Price: {props.service.cost}</Typography>
                    <Typography>
                        Duration: {formatDuration(props.service.duration)}
                    </Typography>
                    <Button>Book</Button>
                </div>
            ) : null}
        </Card>
    );
};

export default ServiceCard;
