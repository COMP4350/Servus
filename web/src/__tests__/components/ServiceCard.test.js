import React from 'react';
import { render } from '@testing-library/react';
import ServiceCard from '../../components/ServiceCard';
import { mockService } from '../../__mocks__/getMockService';

describe('ServiceCard', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <ServiceCard
                key={1}
                service={mockService}
                index={1}
                bg={{ backgroundColor: '#647AA3' }}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
