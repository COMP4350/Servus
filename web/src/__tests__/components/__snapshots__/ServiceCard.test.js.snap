// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-cardView-1 MuiPaper-outlined MuiPaper-rounded"
  >
    <p
      class="MuiTypography-root makeStyles-title-2 MuiTypography-body1 MuiTypography-alignLeft"
    >
      Risto's Painting
    </p>
    <div />
  </div>
</DocumentFragment>
`;
